AK/JsonValue.h

```diff
@@ -92,29 +92,29 @@ class JsonValue {
         return serialized<StringBuilder>();
     }
 
-    int to_int(int default_value = 0) const { return to_i32(default_value); }
-    i32 to_i32(i32 default_value = 0) const { return to_number<i32>(default_value); }
-    i64 to_i64(i64 default_value = 0) const { return to_number<i64>(default_value); }
+    Optional<int> get_int() const { return get_integer<int>(); }
+    Optional<i32> get_i32() const { return get_integer<i32>(); }
+    Optional<i64> get_i64() const { return get_integer<i64>(); }
 
-    unsigned to_uint(unsigned default_value = 0) const { return to_u32(default_value); }
-    u32 to_u32(u32 default_value = 0) const { return to_number<u32>(default_value); }
-    u64 to_u64(u64 default_value = 0) const { return to_number<u64>(default_value); }
-    float to_float(float default_value = 0) const { return to_number<float>(default_value); }
-    double to_double(double default_value = 0) const { return to_number<double>(default_value); }
+    Optional<unsigned> get_uint() const { return get_integer<unsigned>(); }
+    Optional<u32> get_u32() const { return get_integer<u32>(); }
+    Optional<u64> get_u64() const { return get_integer<u64>(); }
+    Optional<float> get_float_with_precision_loss() const { return get_number_with_precision_loss<float>(); }
+    Optional<double> get_double_with_precision_loss() const { return get_number_with_precision_loss<double>(); }
 
-    FlatPtr to_addr(FlatPtr default_value = 0) const
+    Optional<FlatPtr> get_addr() const
     {
 #ifdef __LP64__
```

Comment:
Future-note: We could use an if constexpr (sizeof(FlatPtr) == 4) check for this instead of checking __LP64__